cartography/intel/aws/ec2/instances.py

```diff
@@ -42,17 +42,50 @@
 )
 
 
+@timeit
+def get_ec2_images(boto3_session: boto3.session.Session, image_ids: List[str], region: str) -> Dict[str, Dict]:
+    client = boto3_session.client("ec2", region_name=region, config=get_botocore_config())
+    image_details = {}
+
+    for i in range(0, len(image_ids), 1000):
+        batch = image_ids[i:i+1000]
+
+        try:
+            response = client.describe_images(ImageIds=batch)
+            images = response.get("Image", [])
+            image_details.update({image['ImageId']: image for image in images})
+        except ClientError as e:
+            logger.error(f"Error fetching image details for batch {i//1000 + 1}: {e}")
+            continue
+
+    return image_details
+
+
 @timeit
 @aws_handle_regions
 def get_ec2_instances(boto3_session: boto3.session.Session, region: str) -> List[Dict]:
     client = boto3_session.client("ec2", region_name=region, config=get_botocore_config())
     reservations = []
+    image_ids = []
     try:
         paginator = client.get_paginator("describe_instances")
         for page in paginator.paginate():
             reservations.extend(page["Reservations"])
+            for reservation in page["Reservations"]:
+                for instance in reservation["Instances"]:
+                    image_id = instance.get("ImageId")
+                    if image_id:
+                        image_ids.append(image_id)
+
+        image_details = get_ec2_images(boto3_session, list(set(image_ids)), region)
```

Comment:
let's declare this variable outside try/catch. if for some reason there's an exception, then while accessing the `image_details` variable, it will throw an error.

Reply:
ignore.